File: trunk/java/UMLGraph/src/org/daisy/urakawa/media/timing/TimeOffsetIsNegativeException.java
package org.daisy.urakawa.media.timing;

/**
 * <p>
 * This exception is raised when trying to use a time offset that is negative
 * when it should not be.
 * </p>
 */
public class TimeOffsetIsNegativeException extends
		TimeOffsetIsOutOfBoundsException {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1950794547262877014L;
}

File: exception/TimeOffsetIsNegativeException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// This exception should be raised when trying to use a time offset that is not allowed to be negative.
    /// </summary>
    public class TimeOffsetIsNegativeException : CheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        public TimeOffsetIsNegativeException(string msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        /// <param name="inner">The inner exception</param>
        public TimeOffsetIsNegativeException(string msg, Exception inner)
            : base(msg, inner)
        {
        }
    }
}